src/source/rtl_text_plugin.js

```diff
@@ -1,51 +1,38 @@
 // @flow
-
-const ajax = require('../util/ajax');
 const Evented = require('../util/evented');
-const window = require('../util/window');
 
 let pluginRequested = false;
-let pluginBlobURL = null;
+let pluginURL = null;
 
 module.exports.evented = new Evented();
 
 type ErrorCallback = (error: Error) => void;
 
 module.exports.registerForPluginAvailability = function(
-    callback: (args: {pluginBlobURL: string, errorCallback: ErrorCallback}) => void
+    callback: (args: {pluginURL: string, errorCallback: ErrorCallback}) => void
 ) {
-    if (pluginBlobURL) {
-        callback({ pluginBlobURL: pluginBlobURL, errorCallback: module.exports.errorCallback});
+    if (pluginURL) {
+        callback({ pluginURL: pluginURL, errorCallback: module.exports.errorCallback});
     } else {
         module.exports.evented.once('pluginAvailable', callback);
     }
     return callback;
 };
 
-// Exposed so it can be stubbed out by tests
-module.exports.createBlobURL = function(response: Object) {
-    return window.URL.createObjectURL(new window.Blob([response.data], {type: "text/javascript"}));
-};
 // Only exposed for tests
 module.exports.clearRTLTextPlugin = function() {
     pluginRequested = false;
-    pluginBlobURL = null;
+    pluginURL = null;
 };
 
-module.exports.setRTLTextPlugin = function(pluginURL: string, callback: ErrorCallback) {
+module.exports.setRTLTextPlugin = function(url: string, callback: ErrorCallback) {
     if (pluginRequested) {
         throw new Error('setRTLTextPlugin cannot be called multiple times.');
     }
     pluginRequested = true;
+    pluginURL = url;
     module.exports.errorCallback = callback;
-    ajax.getArrayBuffer({ url: pluginURL }, (err, response) => {
-        if (err) {
-            callback(err);
-        } else if (response) {
-            pluginBlobURL = module.exports.createBlobURL(response);
-            module.exports.evented.fire('pluginAvailable', { pluginBlobURL: pluginBlobURL, errorCallback: callback });
-        }
-    });
+    module.exports.evented.fire('pluginAvailable', { pluginURL: pluginURL, errorCallback: callback });
```

Comment:
Since we're no longer requesting the URL here, a malformed URL that would previously have resulted in an error callback would now just get passed along to the worker (as seen int he "some bogus url" unit test below).  Now that we don't have the AJAX request effectively validating the url for us, should we do so directly here?

Reply:
We do still feed errors from the workers to the foreground, but the caller needs to pass in an error callback and while that's documented we don't actually pass a callback in our example code -- maybe we should just change the example to include a simple logging error callback? The actual loading error will hopefully be more informative than whatever we come up with from URL validation.

Reply:
Ah, I see -- yeah, that seems fine